### ISSUE TITLE
Getting Build issue after installing and using "Angular-auth-oidc-client"

### ISSUE THREAD

**Issue by SampatPenugonda** [damienbod/angular-auth-oidc-client#457]
After installing and importing into app.module.ts, we are getting build issues [ during ng serve ]. 

![image](https://user-images.githubusercontent.com/19400831/59729759-85728700-925d-11e9-9a65-bd0e8bde50e9.png)

Do we need to install any dependencies ? 

Here are my Package.json file

{
  "name": "angular6-example",
  "version": "0.0.0",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "^6.0.2",
    "@angular/cdk": "^7.1.1",
    "@angular/common": "^6.0.2",
    "@angular/compiler": "^6.0.2",
    "@angular/core": "^6.0.2",
    "@angular/forms": "^6.0.2",
    "@angular/http": "^6.0.2",
    "@angular/platform-browser": "^6.0.2",
    "@angular/platform-browser-dynamic": "^6.0.2",
    "@angular/router": "^6.0.2",
    "@progress/kendo-angular-buttons": "^4.3.2",
    "@progress/kendo-angular-dateinputs": "2 - 3",
    "@progress/kendo-angular-dialog": "^3.10.1",
    "@progress/kendo-angular-dropdowns": "2 - 3",
    "@progress/kendo-angular-excel-export": "1 - 2",
    "@progress/kendo-angular-grid": "^3.14.0",
    "@progress/kendo-angular-inputs": "2 - 3",
    "@progress/kendo-angular-intl": "^1.0.0",
    "@progress/kendo-angular-l10n": "^1.3.0",
    "@progress/kendo-angular-layout": "^3.2.0",
    "@progress/kendo-angular-notification": "^1.2.0",
    "@progress/kendo-angular-popup": "^2.0.0",
    "@progress/kendo-angular-tooltip": "^1.1.2",
    "@progress/kendo-data-query": "^1.0.0",
    "@progress/kendo-drawing": "^1.0.0",
    "@progress/kendo-theme-default": "latest",
    "@slb-dls/angular": "^6.1.3",
    "@slb-dls/vanilla": "^4.1.6",
    "angular-auth-oidc-client": "^10.0.0",
    "angular2-moment": "^1.9.0",
    "angularx-flatpickr": "^6.1.0",
    "babel-polyfill": "^6.26.0",
    "bootstrap": "^3.4.0",
    "classlist.js": "^1.1.20150312",
    "core-js": "^2.5.4",
    "flatpickr": "^4.5.2",
    "jquery": "^3.3.1",
    "oidc-client": "^1.7.1",
    "omni-slider": "^1.2.1",
    "pikaday": "^1.8.0",
    "rxjs": "^6.0.0",
    "rxjs-compat": "^6.0.0",
    "taggle": "^1.14.1",
    "web-animations-js": "^2.3.1",
    "zone.js": "^0.8.26"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "^0.13.8",
    "@angular/cli": "~6.0.3",
    "@angular/compiler-cli": "^6.0.2",
    "@angular/language-service": "^6.0.2",
    "@types/jasmine": "~2.8.6",
    "@types/jasminewd2": "~2.0.3",
    "@types/node": "~8.9.4",
    "codelyzer": "~4.2.1",
    "jasmine-core": "~2.99.1",
    "jasmine-spec-reporter": "~4.2.1",
    "karma": "~1.7.1",
    "karma-chrome-launcher": "~2.2.0",
    "karma-coverage-istanbul-reporter": "~1.4.2",
    "karma-jasmine": "~1.1.1",
    "karma-jasmine-html-reporter": "^0.2.2",
    "protractor": "~5.3.0",
    "ts-node": "~5.0.1",
    "tslint": "~5.9.1",
    "typescript": "~2.7.2"
  }
}

**Comment by damienbod** [comment]
You could compare this this:


https://github.com/damienbod/dotnet-template-angular/blob/master/dotnet-angular-azure-ad-oidc/ClientApp/package.json

Greetings Damien

**Comment by SampatPenugonda** [comment]
Hi @damienbod , 

I can't upgrade my project to angular 8, is there any specific version of this supports angular 6 ?

Thanks, 
Sampath Penugonda

**Comment by FabianGosebrink** [comment]
Did you migrate to the latest configuration model here? Check the docs please.

```
 const config: OpenIdConfiguration = {
                stsServer: configResult.customConfig.stsServer,
                redirect_url: 'https://localhost:4200',
                client_id: 'angularClient',
                scope: 'openid profile email',
                response_type: 'code',
                silent_renew: true,
                silent_renew_url: 'https://localhost:4200/silent-renew.html',
                log_console_debug_active: true,
                // all other properties you want to set
            };
```

**Comment by SampatPenugonda** [comment]
I have installed v 9.0.8 this is supporting angular 6, v 10.0.0 required angular 8 and typescript version 3.x.x, I can't upgrade my application to Angular 8, so I am using an earlier version of this package.

**Comment by FabianGosebrink** [comment]
But did you upgrade to the latest config model? https://github.com/damienbod/angular-auth-oidc-client/issues/457#issuecomment-503480243

**Comment by gbaquedano** [comment]
I think he's right, it doesn't have to do with config model. V10.0.0 won't work with previous versions of Angular 8 which I guess is the version it was compiled with.

Got this warnings:
./node_modules/angular-auth-oidc-client/fesm5/angular-auth-oidc-client.js 198:58-76
"export 'ɵɵdefineInjectable' was not found in '@angular/core'
./node_modules/angular-auth-oidc-client/fesm5/angular-auth-oidc-client.js 319:63-81
"export 'ɵɵdefineInjectable' was not found in '@angular/core'
./node_modules/angular-auth-oidc-client/fesm5/angular-auth-oidc-client.js 198:154-162
"export 'ɵɵinject' was not found in '@angular/core'
./node_modules/angular-auth-oidc-client/fesm5/angular-auth-oidc-client.js 319:169-177
"export 'ɵɵinject' was not found in '@angular/core'

So I believe IVY was used, because according to this commit, ɵɵinject got deprecated and removed before v8.0.0 release:
https://github.com/angular/angular/commit/4f9b16783b838f03793a33eed39dd6c8112519d9

**Comment by FabianGosebrink** [comment]
Hey @gbaquedano ,

yeah, we updated to angular 8 but not ivy. Do you have a sample repo to reproduce the issue?

Fabian

**Comment by shwetashukla** [comment]
so which version is compatible with Angular 7

**Comment by gbaquedano** [comment]
@FabianGosebrink, here we go.

`
ng --version

Angular CLI: 7.3.9
Node: 10.16.0
OS: win32 x64
Angular: 7.2.15`
`
Creating minimal project.
`
ng new v7-v10 --skipTests=true --style=scss --skipGit=true --routing=true --minimal=true --skipInstall=true
`
Package.json
`
{
  "name": "v7-v10",
  "version": "0.0.0",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "~7.2.15",
    "@angular/common": "~7.2.15",
    "@angular/compiler": "~7.2.15",
    "@angular/core": "~7.2.15",
    "@angular/forms": "~7.2.15",
    "@angular/platform-browser": "~7.2.15",
    "@angular/platform-browser-dynamic": "~7.2.15",
    "@angular/router": "~7.2.15",
    "angular-auth-oidc-client": "^10.0.0",
    "core-js": "~2.6.9",
    "rxjs": "~6.5.2",
    "tslib": "^1.10.0",
    "zone.js": "~0.8.29"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "~0.13.9",
    "@angular/cli": "~7.3.9",
    "@angular/compiler-cli": "~7.2.15",
    "@angular/language-service": "~7.2.15",
    "@types/node": "~10.14.9",
    "ts-node": "^8.3.0",
    "tslint": "^5.17.0",
    "typescript": "~3.1.6"
  }
}
`
Edit src/app/app.module.ts with:
`
import { BrowserModule } from '@angular/platform-browser';
import { NgModule, APP_INITIALIZER } from '@angular/core';

import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';

import { AuthModule, ConfigResult, OidcConfigService, OidcSecurityService, OpenIdConfiguration } from 'angular-auth-oidc-client';

export function loadConfig(oidcConfigService: OidcConfigService) {
    return () => oidcConfigService.load_using_stsServer('http://localhost:4200');
}

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule,
    AppRoutingModule
  ],
    providers: [
        OidcConfigService,
        {
            provide: APP_INITIALIZER,
            useFactory: loadConfig,
            deps: [OidcConfigService],
            multi: true,
        },],
  bootstrap: [AppComponent]
})
export class AppModule {
    constructor(private oidcSecurityService: OidcSecurityService, private oidcConfigService: OidcConfigService) {
        this.oidcConfigService.onConfigurationLoaded.subscribe((configResult: ConfigResult) => {

            // Use the configResult to set the configurations

            const config: OpenIdConfiguration = {
                stsServer: configResult.customConfig.stsServer,
                redirect_url: 'https://localhost:4200',
                client_id: 'angularClient',
                scope: 'openid profile email',
                response_type: 'code',
                silent_renew: true,
                silent_renew_url: 'https://localhost:4200/silent-renew.html',
                log_console_debug_active: true,
                // all other properties you want to set
            };

            this.oidcSecurityService.setupModule(config, configResult.authWellknownEndpoints);
        });
    }
}
`
Run the app and check output:
`
ng serve --open

...

Date: 2019-06-20T12:17:48.181Z - Hash: 047eb7acc8dad955850f - Time: 5758ms
4 unchanged chunks
chunk {main} main.js, main.js.map (main) 12.7 kB [initial] [rendered]
chunk {vendor} vendor.js, vendor.js.map (vendor) 4.25 MB [initial] [rendered]

WARNING in ./node_modules/angular-auth-oidc-client/fesm5/angular-auth-oidc-client.js 198:58-76
"export 'ɵɵdefineInjectable' was not found in '@angular/core'

WARNING in ./node_modules/angular-auth-oidc-client/fesm5/angular-auth-oidc-client.js 319:63-81
"export 'ɵɵdefineInjectable' was not found in '@angular/core'

WARNING in ./node_modules/angular-auth-oidc-client/fesm5/angular-auth-oidc-client.js 198:154-162
"export 'ɵɵinject' was not found in '@angular/core'

WARNING in ./node_modules/angular-auth-oidc-client/fesm5/angular-auth-oidc-client.js 319:169-177
"export 'ɵɵinject' was not found in '@angular/core'
i ｢wdm｣: Compiled with warnings.

`

**Comment by FabianGosebrink** [comment]
Okay, we will have a look at that. Please provide a repository to clone next time. Thanks.

**Comment by damienbod** [comment]
Here's the 9.0.8 documentation

https://github.com/damienbod/angular-auth-oidc-client/tree/relase_9_0_8

This seems strange that Angular 7 has problems with this. I have moved all my projects to 8. I'll look into this

Greetings Damien

**Comment by SampatPenugonda** [comment]
> Here's the 9.0.8 documentation
> 
> https://github.com/damienbod/angular-auth-oidc-client/tree/relase_9_0_8
> 
> This seems strange that Angular 7 has problems with this. I have moved all my projects to 8. I'll look into this
> 
> Greetings Damien

I am using v 9.0.8, it's compatible with Angular 6. I am new to this package, trying to under available functionality. 

Any detailed docs are available?

**Comment by SampatPenugonda** [comment]
By the way, where the token will be stored? Is it in-memory? I can see the token in the browser, but not able to fetch the token value by using _securityService.getToken() it is returning an empty/undefined value.

**Comment by damienbod** [comment]
@SampatPenugonda  fixed in version 10.0.1
